azurerm/internal/services/databricks/databricks_workspace_resource.go

```diff
@@ -79,46 +79,98 @@ func resourceDatabricksWorkspace() *pluginsdk.Resource {
-								"custom_parameters.0.private_subnet_name", "custom_parameters.0.virtual_network_id",
+							Computed: true,
+							MaxItems: 1,
+							Elem: &pluginsdk.Resource{
+								Schema: map[string]*pluginsdk.Schema{
+									"key_source": {
+										Type:             pluginsdk.TypeString,
+										Optional:         true,
+										Default:          "Default",
+										DiffSuppressFunc: suppress.CaseDifference,
+										ValidateFunc: validation.StringInSlice([]string{
+											string(databricks.Default),
+											string(databricks.MicrosoftKeyvault),
+										}, true),
+									},
+									"key_name": {
+										Type:     pluginsdk.TypeString,
+										Optional: true,
+									},
+									"key_version": {
+										Type:     pluginsdk.TypeString,
+										Optional: true,
+									},
+									"key_vault_uri": {
+										Type:     pluginsdk.TypeString,
+										Optional: true,
+									},
+								},
 							},
+							AtLeastOneOf: workspaceCustomParametersString(),
+						},
+
+						"no_public_ip": {
+							Type:         pluginsdk.TypeBool,
+							ForceNew:     true,
+							Optional:     true,
+							Default:      false,
+							AtLeastOneOf: workspaceCustomParametersString(),
 						},
 
 						"public_subnet_name": {
-							Type:     pluginsdk.TypeString,
-							ForceNew: true,
-							Optional: true,
-							AtLeastOneOf: []string{"custom_parameters.0.no_public_ip", "custom_parameters.0.public_subnet_name",
-								"custom_parameters.0.private_subnet_name", "custom_parameters.0.virtual_network_id",
-							},
+							Type:         pluginsdk.TypeString,
+							ForceNew:     true,
+							Optional:     true,
+							AtLeastOneOf: workspaceCustomParametersString(),
 						},
 
 						"private_subnet_name": {
-							Type:     pluginsdk.TypeString,
-							ForceNew: true,
-							Optional: true,
-							AtLeastOneOf: []string{"custom_parameters.0.no_public_ip", "custom_parameters.0.public_subnet_name",
-								"custom_parameters.0.private_subnet_name", "custom_parameters.0.virtual_network_id",
-							},
+							Type:         pluginsdk.TypeString,
+							ForceNew:     true,
+							Optional:     true,
+							AtLeastOneOf: workspaceCustomParametersString(),
 						},
 
 						"virtual_network_id": {
 							Type:         pluginsdk.TypeString,
 							ForceNew:     true,
 							Optional:     true,
 							ValidateFunc: azure.ValidateResourceIDOrEmpty,
-							AtLeastOneOf: []string{"custom_parameters.0.no_public_ip", "custom_parameters.0.public_subnet_name",
-								"custom_parameters.0.private_subnet_name", "custom_parameters.0.virtual_network_id",
-							},
+							AtLeastOneOf: workspaceCustomParametersString(),
+						},
+
+						"enable_cmk_encryption": {
+							Type:         pluginsdk.TypeBool,
+							Optional:     true,
+							Default:      false,
+							AtLeastOneOf: workspaceCustomParametersString(),
+						},
+
+						"enable_infrastructure_encryption": {
```

Comment:
could we change this to match the rest of the provideR?
```suggestion
						"infrastructure_encryption_enabled": {
```